x-pack/test/fleet_api_integration/apis/enrollment_api_keys/crud.ts

```diff
@@ -115,6 +115,55 @@ export default function (providerContext: FtrProviderContext) {
         expect(apiResponse.item).to.have.keys('id', 'api_key', 'api_key_id', 'name', 'policy_id');
       });
 
+      it('should allow to create an enrollment api key with agent policy and unique name', async () => {
+        const { body: noSpacesRes } = await supertest
+          .post(`/api/fleet/enrollment-api-keys`)
+          .set('kbn-xsrf', 'xxx')
+          .send({
+            policy_id: 'policy1',
+            name: 'something',
+          });
+        expect(noSpacesRes.item).to.have.keys('id', 'api_key', 'api_key_id', 'name', 'policy_id');
+
+        const { body: hasSpacesRes } = await supertest
+          .post(`/api/fleet/enrollment-api-keys`)
+          .set('kbn-xsrf', 'xxx')
+          .send({
+            policy_id: 'policy1',
+            name: 'something else',
+          });
+        expect(hasSpacesRes.item).to.have.keys('id', 'api_key', 'api_key_id', 'name', 'policy_id');
+
+        const { body: noSpacesDupe } = await supertest
+          .post(`/api/fleet/enrollment-api-keys`)
+          .set('kbn-xsrf', 'xxx')
+          .send({
+            policy_id: 'policy1',
+            name: 'something',
+          })
+          .expect(500);
+
+        expect(noSpacesDupe).to.eql({
+          statusCode: 500,
+          error: 'Internal Server Error',
+          message: 'An enrollment key named something already exists for agent policy policy1',
+        });
+
+        const { body: hasSpacesDupe } = await supertest
+          .post(`/api/fleet/enrollment-api-keys`)
+          .set('kbn-xsrf', 'xxx')
+          .send({
+            policy_id: 'policy1',
+            name: 'something else',
+          })
+          .expect(500);
+        expect(hasSpacesDupe).to.eql({
+          statusCode: 500,
```

Comment:
nitpick and probably not to solve in this PR, but this seems more a user error to me and should have a 400 no?

Reply:
Totally agree. I replaced this `Error` with `IngestManagerError` and then found myself wanting to replace the others in this file and I stopped/reverted.

I'll update this one since it has a test now and we can deal with the others later.

Reply:
Added in https://github.com/elastic/kibana/pull/100056/commits/931f54a11b183f156e6573034bd3145f2402fe77